src/SOS/Strike/strike.cpp

```diff
@@ -16713,6 +16713,23 @@ DECLARE_API(SetSymbolServer)
         DisableSymbolStore();
     }
 
+    if (searchDirectory.data != nullptr) {
+        std::string fullPath;
+        if (!GetAbsolutePath(searchDirectory.data, fullPath))
+        {
+            ExtErr("Invalid search directory %s\n", fullPath.c_str());
+            return E_FAIL;
+        }
+
+        size_t length = fullPath.length() + 1;
+        delete[] searchDirectory.data;
+        searchDirectory.data = new NOTHROW char[length];
+        if (searchDirectory.data)
+        {
+            strcpy_s(searchDirectory.data, length, fullPath.c_str());
+        }
```

Comment:
Does this need to handle the OOM case? CC @mikem8361 since I don't know the convention in SOS.

Reply:
Yes, I think it is good to E_FAIL in else case, I will add.

Reply:
I'd rather you not have to do this allocation and failure path.